ultradns-rest/src/main/java/denominator/ultradns/service/UltraDNSRestZoneApi.java

```diff
@@ -0,0 +1,179 @@
+package denominator.ultradns.service;
+
+import denominator.model.Zone;
+import denominator.ultradns.service.integration.UltraDNSRest;
+import denominator.ultradns.exception.UltraDNSRestException;
+import denominator.ultradns.model.AccountList;
+import denominator.ultradns.model.RRSet;
+import denominator.ultradns.model.Record;
+import denominator.ultradns.util.RRSetUtil;
+import denominator.ultradns.util.ZoneUtil;
+import org.apache.commons.lang.StringUtils;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import static denominator.common.Preconditions.checkState;
+import static denominator.common.Util.singletonIterator;
+import denominator.ResourceTypeToValue.ResourceTypes;
+import org.apache.log4j.Logger;
+
+public final class UltraDNSRestZoneApi implements denominator.ZoneApi {
+
+  private final UltraDNSRest api;
+
+  @Inject
+  UltraDNSRestZoneApi(UltraDNSRest api) {
+    this.api = api;
+  }
+
+  private static final Logger LOGGER = Logger.getLogger(UltraDNSRestZoneApi.class);
+
+  /**
+   * in UltraDNSRest, zones are scoped to an account.
+   */
+  @Override
+  public Iterator<Zone> iterator() {
+    final String accountName = getCurrentAccountName();
```

Comment:
Yes, the UltraDNS SOAP module is getting the list of zone for a particular account. We followed the same logic.